internal/db/storage/blob/s3/s3_test.go

```diff
@@ -50,42 +64,133 @@ func TestNew(t *testing.T) {
+
+		opts := []comparator.Option{
+			comparator.AllowUnexported(client{}, aws.Config{}),
+			comparator.IgnoreTypes(request.Handlers{}),
+			comparator.Comparer(func(want, got errgroup.Group) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got aws.Config) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got *session.Session) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got reader.Reader) bool {
+				return want != nil && got != nil
+			}),
+			comparator.Comparer(func(want, got writer.Writer) bool {
+				return want != nil && got != nil
+			}),
+			comparator.Comparer(func(want, got []backoff.Option) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got s3manager.S3Manager) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+		}
+
+		if diff := comparator.Diff(w.want, got, opts...); len(diff) != 0 {
+			return errors.Errorf("diff: %s", diff)
 		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           opts: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           opts: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			return test{
+				name: "returns error when option is empty and session field is nil",
+				want: want{
+					want: nil,
+					err:  errors.NewErrInvalidOption("session", nil),
+				},
+			}
+		}(),
+
+		func() test {
+			opt := func(c *client) error {
+				return errors.New("err")
+			}
+			return test{
+				name: "returns error when option apply fails",
+				args: args{
+					opts: []Option{
+						opt,
+					},
+				},
+				want: want{
+					want: nil,
+					err:  errors.ErrOptionFailed(errors.New("err"), reflect.ValueOf(opt)),
+				},
+			}
+		}(),
+
+		func() test {
+			sess, _ := session.NewSession()
+			r := new(reader.MockReader)
+			w := new(writer.MockWriter)
+			return test{
+				name: "returns bucket and nil when the option apply success and no error occurs internally",
+				args: args{
+					opts: []Option{
+						WithSession(sess),
+						WithReader(r),
+						WithWriter(w),
+					},
+				},
+				want: want{
+					want: &client{
+						eg:      errgroup.Get(),
+						session: sess,
+						service: s3.New(sess),
+						reader:  r,
+						writer:  w,
+					},
+					err: nil,
+				},
+			}
+		}(),
+
+		func() test {
+			sess, _ := session.NewSession()
+			service := s3.New(sess)
+
+			eg := errgroup.Get()
+
+			return test{
+				name: "returns bucket and nil when reader and writer are created and no error occurs internally",
```

Comment:
it would be better for robust test when this test have these additional patterns:
- `returns bucket and nil when writer is created and no error occurs internally`
- `returns bucket and nil when reader is created and no error occurs internally`

Reply:
What did you do about this comment?

Reply:
I created two test case based on your comment.
I am sorry, if I made a mistake.

https://github.com/vdaas/vald/blob/34d8fb7e8f7da468d4dfe92a81293ec188ea6c82/internal/db/storage/blob/s3/s3_test.go#L151-L230

Reply:
sorry, I misread it. 
so I will fix it soon.

Reply:
> it would be better for robust test when this test have these additional patterns:

I think so too. thank you